apps/web/src/hooks/useVariablesManager.ts

```diff
@@ -1,64 +1,64 @@
-import { useEffect, useMemo, useState } from 'react';
+import { useCallback, useLayoutEffect, useState } from 'react';
 import { useFieldArray, useFormContext } from 'react-hook-form';
 import { parse } from '@handlebars/parser';
+import isEqual from 'lodash.isequal';
 import { getTemplateVariables } from '@novu/shared';
 
-export const useVariablesManager = (index: number, contents: string[]) => {
-  const [ast, setAst] = useState<any>({ body: [] });
-  const [textContent, setTextContent] = useState<string>('');
-  const { watch, control, getValues } = useFormContext();
+import type { IForm, ITemplates } from '../components/templates/formTypes';
 
+export const useVariablesManager = (index: number, contents: string[]) => {
+  const { watch, control, getValues } = useFormContext<IForm>();
   const variablesArray = useFieldArray({ control, name: `steps.${index}.template.variables` });
   const variableArray = watch(`steps.${index}.template.variables`, []);
 
-  useEffect(() => {
+  const getTextContent = useCallback((templateToParse?: ITemplates): string => {
+    return contents
+      .map((con) => con.split('.').reduce((a, b) => a && a[b], templateToParse ?? {}))
+      .map((con) => (Array.isArray(con) ? con.map((innerCon) => innerCon.content).join(' ') : con))
+      .join(' ');
+  }, []);
+
+  const [textContent, setTextContent] = useState<string>(() => getTextContent(getValues(`steps.${index}.template`)));
```

Comment:
Calculate the `textContent` string variable that holds all the template field values which might use variables - on the first render.